ir/function.h

```diff
@@ -171,4 +171,17 @@ class CFG final {
   void printDot(std::ostream &os) const;
 };
 
+// Does not support loops
+class DomTree final {
+    Function& f;
+    CFG& cfg;
+    // src -> vec<bb's that dominate src in order of their depth in the CFG>
+    std::unordered_map<const BasicBlock*,
+                      std::vector<const BasicBlock*>> dominators;
```

Comment:
including the BB in its own list is probably not very interesting.
Also, do we need to keep all dominators or only the immediate one?

Reply:
The most important dominator is the immediate one which will always be the before last dominator in the vector and i don't think i need any other for the formulas algorithm. However we get all of them as part of the construction because to build dominators for x it needs to intersect the dominators of it's predecessors.

Reply:
Including the BB itself is part of the algorithm for it to work, it could be removed at the end.